lib/Driver/Compilation.cpp

```diff
@@ -102,7 +102,11 @@ Compilation::Compilation(DiagnosticEngine &Diags,
                          std::unique_ptr<InputArgList> InputArgs,
                          std::unique_ptr<DerivedArgList> TranslatedArgs,
                          InputFileList InputsWithTypes,
-                         StringRef ArgsHash, llvm::sys::TimePoint<> StartTime,
+                         std::string CompilationRecordPath,
+                         bool OutputCompilationRecordForModuleOnlyBuild,
+                         StringRef ArgsHash,
+                         llvm::sys::TimePoint<> StartTime,
+                         llvm::sys::TimePoint<> LastBuildTime,
```

Comment:
One inevitable trend that happens to code over time is that classes get larger with more instance variables, just like how functions tend to grow with more lines of code. One way to improve the clarity of this change might be to figure out if it makes sense to group these new instance variables into a new class that would be included as a member of this class.
Does that make sense here?

Reply:
I'll see grouping which fields into a class makes sense.